### ISSUE TITLE
Modules needed

### ISSUE THREAD

**Issue by alesrosso** [cmotc/android_product_antm_revo#1]
Hi, I have a chinese tablet (Sapporo) and your modules worked on. So I hope there is some way to get the .config to make my own modules that isn't are here? Or some way to help me on my own modules compile, with this configuration?

Sorry about my english plz!!

Best regards
Alejandro

**Comment by cmotc** [comment]
Absolutely. I've been really busy in the past few weeks, and some things have changed(They've even been updated on the sunxi wiki) so obviously your mileage may vary, but the last known good config I have is still available and I can probably help you figure out how to make the changes you need. I am pretty much an amateur though.

As far as the last config I had I knew worked, it's here: https://raw.githubusercontent.com/cmotc/fireaxe-kernel-grsec/master/arch/arm/configs/antm_seven_working, that said, it's got the PAX/GRSecurity patches enabled which might make it a little harder to work with at first. If you remove all the lines containing CONFIG_GRKERNSEC and do "make olddefconfig" then that might get you pretty close. But the truth is all I did was make a modified version sunxi_defconfig using the regular tools for doing so.

The process is basically that of identifying hardware, and enabling drivers and kernel features required for it to function. So what you do is export your architecture and if necessary your cross-compiler prefix, then you copy arch/arm/configs/sunxi_defconfig to .config, and then you do "make nconfig" and find the necessary options. In the case of an A33 tablet, you'll need to enable Frame Buffer based graphics, turn on the blacklight, enable wireless drivers, and a handful of other things along those lines. Sometimes finding the option itself is tedious, I use this as a shortcut by putting it in my bash_aliases and searching within the Kconfig files for terms related to the hardware you have. I'm not sure this is the best way to do it, but it works for me.

        searchkconfig(){
            SEARCH_WORD=$@
            if [ -z "$SEARCH_WORD" ]; then
                for file in $(find . -name Kconfig); do cat $file | grep -A 9 -i $SEARCH_WORD; done
            fi
        }

So, if I wanted to find the option required to enable the RTL8187 wireless chip, then you might do "searchkconfig rtl8187" from the kernel directory. Let's see... I'm trying to make sure I give you whatever help I can but like I said, I'm a self-taught amateur and I'm also trying to make sure I don't steer you wrong. Let me know if I can give you any more specific help. I try to get back to people as quick as I can but sometimes I'm busy for long times.

**Comment by cmotc** [comment]
There's also this. In here, there are debs of various kernels I have experimented with on my tablet and a script that makes an img file that, at this point, usually boots. You might try using it/them too. https://github.com/cmotc/imgmaker
